src/hotspot/os/posix/os_posix.cpp

```diff
@@ -610,9 +610,14 @@ void os::print_jni_name_suffix_on(outputStream* st, int args_size) {
 
 bool os::get_host_name(char* buf, size_t buflen) {
   struct utsname name;
-  uname(&name);
-  jio_snprintf(buf, buflen, "%s", name.nodename);
-  return true;
+  int retcode = uname(&name);
+  if (retcode != -1) {
+    jio_snprintf(buf, buflen, "%s", name.nodename);
+    return true;
+  }
+  const char* errmsg = os::strerror(retcode);
```

Comment:
That should be `errno` not `retcode`.

Reply:
Of course. Thank you for noticing and commenting.